### ISSUE TITLE
Dupicated programms with pvr.hts and pvr.dvbviewer and Kodi 19

### ISSUE THREAD

**Issue by enigma131** [xbmc/xbmc#19345]
### Describe the bug
I use boths pvr.hts and pvr.dvbviewer activated on all client devices because server is some time under windows ans some time under Linux. Boths pvr are up to date 8.2.3 for first and 7.2.0 for second

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
Have only the programm lit list of active pvr (because only 1 can reach the server) . This was the working mode for kodi 18

## Actual Behavior
Have programms in double

## Possible Fix
cache problem ?

### To Reproduce
1. Start the server under Windows with Windows media server
2. Start one client , get the programm list,watch,  then shut down the client
3. Reboot the server under linux with tvheadend
4. Start the client again, and observe what happening for programm list is t in double

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ x] Android 10
 - [ ] iOS
 - [ ] tvOS
 - [x ] Linux 
 - [ ] OSX
 - [x ] Windows 10
 - [ ] Windows UWP

 - Operating system version/name: all of above
 - Kodi version: 19.0 final

I provide logs in 30 minutes

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by phunkyfish** [comment]
So both servers deliver the same channels but you would only run one at a time?

In this case in kodi 19 because the EPG is stored in a local DB you will have both sets of data. The only way this would clear would be to have the server serve no channels so that kodi clears its DB (of channels and EPG data) before switching.

**Comment by ksooo** [comment]
Not sure this use related to EPG database @phunkyfish

**Comment by enigma131** [comment]
I have added log and pictures . No only 1 sever at a time, it is always same computer

**Comment by phunkyfish** [comment]
> Not sure this is related to EPG database @phunkyfish. Channels were also cached in v18... What is new with 19 is that channels from non reachable backends (error on retrival off channels list, for example) are not deleted from Kodi PVR DB. That could be related.

Yes, but now they also still have EPG data after a restart. Remind me again why the channels are not removed if the backend is unavailable?

**Comment by ksooo** [comment]
It is important not to delete (temporarily) stale channels from the DB, because this will also wipe data only available locally, like last played info and link to EPG data gets lost etc. 

Maybe we should hide those channels, but we must not delete them from the DB.

I will take a look.

**Comment by enigma131** [comment]
I've got Kodi version 19.1 , problem solved now! Thanks
